components/editor/prosemirror-plugins.md

```diff
@@ -0,0 +1,134 @@
+| `ProseMirror` | An object that contains various ProseMirror classes and functions.|
+| `getPlugins` | A function that accepts `Schema` as an argument and returns the default Editor plugins. The function must return an array of ProseMirror plugins. |
+
+> To ensure all the built-in functionalities of the Editor are working correctly, the result array must contain the default plugins which can be retrieved by calling the `getPlugins` function.
+
+>caption Adding a Placeholder Plugin
+
+```CSHTML
+@using Telerik.Blazor.Components.Editor
+@inject IJSRuntime js
+
+<script>
+    window.pluginsProvider = (args) => {
+        const schema = args.getSchema();
+        var placeHolderKey = new args.ProseMirror.PluginKey("placeholder");
+
+        function placeholder(emptyMessage) {
+            console.log(args);
+            return new args.ProseMirror.Plugin({
+                key: placeHolderKey,
+                props: {
+                    decorations: (state) => {
+                        const { doc } = state;
+                        const empty =
+                            doc.textContent === "" &&
+                            doc.childCount <= 1 &&
+                            doc.content.size <= 2;
+                        if (!empty) {
+                            return args.ProseMirror.DecorationSet.empty;
+                        }
+                        const decorations = [];
+                        const decAttrs = {
+                            class: "placeholder",
+                            "data-placeholder": emptyMessage,
+                        };
+                        doc.descendants((node, pos) => {
+                            decorations.push(args.ProseMirror.Decoration.node(pos, pos + node.nodeSize, decAttrs));
+                        });
+                        return args.ProseMirror.DecorationSet.create(doc, decorations);
+                    }
+                }
+            });
+        }        
+
+        return [...args.getPlugins(schema), placeholder("Enter some content ...")];
+    }
+
+    function injectEditorStyleTag() {
+        var doc = document.querySelector("iframe").contentWindow.document;
+        var head = doc.querySelector("head");
+
+        var style = doc.createElement("style");
+        style.type = "text/css";
+
+        var iframeStyles = `p.placeholder:first-child:before {
+                        content: attr(data-placeholder);
+                        float: left;
+                        color: rgb(0, 0, 0, 0.5);
+                        cursor: text;
+                        height: 0;
+                        font-style: italic;
+                    }`;
+
+        style.appendChild(doc.createTextNode(iframeStyles));
+        head.appendChild(style);
+    };
+</script>
+
+<TelerikEditor Height="400px"
```

Comment:
I would move the Editor tag to the top. In this way, the reader will see something familiar first, including the Plugins parameter value. This may help the reader to identify bits in the JavaScript code afterwards.

Applies to all similar snippets.